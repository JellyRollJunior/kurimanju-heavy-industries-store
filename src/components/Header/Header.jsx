import logo from '../../assets/logo.svg';
import headerIcon from '../../assets/header-icon.png';
import headerPortrait from '../../assets/header-portrait.png';
import headerHudText from '../../assets/header-hud-text.png';
import styles from './Header.module.css';

const Header = () => {
  return (
    <header className={styles.header}>
      <img src={headerIcon} alt="" className={styles.headerIcon} />
      <img src={logo} alt="" className={styles.logo} />
      <div className={styles.headerDecoration}>
        <p className={styles.chineseText}>
          機体究更
          <br />
          武裝解除
          <br />
          決定中止
        </p>
        <p className={styles.englishText}>
          F91 EARTH-WIND RISES
          <br />
          LIBERATION X-97
          <br />
          DE- 524 - ITY
        </p>
        <div></div>
        <img src={headerPortrait} alt="" className={styles.portrait} />
      </div>
      <img src={headerHudText} alt="" className={styles.hudText} />
    </header>
  );
};

export { Header };
