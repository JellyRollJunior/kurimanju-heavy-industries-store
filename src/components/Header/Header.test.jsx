import { render, screen } from '@testing-library/react';
import { describe, expect, it, vi } from 'vitest';
import { Header } from './Header.jsx';
import userEvent from '@testing-library/user-event';

describe('Header component', () => {
  it('Renders icon, logo, text designs, portrait, and HUD text', () => {
    const { container } = render(<Header />);

    expect(container).toMatchSnapshot();
  });

  it('Calls onClickViewCart when clicking the View Cart button', async () => {
    const onClickViewCart = vi.fn();
    const user = userEvent.setup();
    render(<Header onClickViewCart={onClickViewCart} />);
    const viewCartButton = screen.getByRole('button', { name: 'View Cart' });

    await user.click(viewCartButton);

    expect(onClickViewCart).toBeCalled();
  });
});
