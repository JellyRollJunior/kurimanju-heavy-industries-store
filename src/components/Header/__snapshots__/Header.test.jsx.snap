// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header component > Renders icon, logo, text designs, portrait, and HUD text 1`] = `
<div>
  <header
    class="_header_63c2ea"
  >
    <img
      alt=""
      class="_headerIcon_63c2ea"
      src="/src/assets/header-icon.png"
    />
    <img
      alt=""
      class="_logo_63c2ea"
      src="/src/assets/logo.svg"
    />
    <div
      class="_headerDecoration_63c2ea"
    >
      <p
        class="_chineseText_63c2ea"
      >
        機体究更
        <br />
        武裝解除
        <br />
        決定中止
      </p>
      <p
        class="_englishText_63c2ea"
      >
        F91 EARTH-WIND RISES
        <br />
        LIBERATION X-97
        <br />
        DE- 524 - ITY
      </p>
      <div />
      <img
        alt=""
        class="_portrait_63c2ea"
        src="/src/assets/header-portrait.png"
      />
      <div />
      <button
        class="_viewCartButton_63c2ea"
      >
        View Cart
      </button>
    </div>
    <img
      alt=""
      class="_hudText_63c2ea"
      src="/src/assets/header-hud-text.png"
    />
  </header>
</div>
`;
